rclcpp/src/rclcpp/intra_process_manager.cpp

```diff
@@ -225,5 +225,48 @@ IntraProcessManager::can_communicate(
   return true;
 }
 
+size_t
+IntraProcessManager::lowest_available_capacity(const uint64_t intra_process_publisher_id) const
+{
+  size_t capacity = std::numeric_limits<size_t>::max();
+
+  auto publisher_it = pub_to_subs_.find(intra_process_publisher_id);
+  if (publisher_it == pub_to_subs_.end()) {
+    // Publisher is either invalid or no longer exists.
+    RCLCPP_WARN(
+      rclcpp::get_logger("rclcpp"),
+      "Calling lowest_available_capacity for invalid or no longer existing publisher id");
+    return 0;
+  }
+
+  for (const auto sub_id : publisher_it->second.take_shared_subscriptions) {
+    capacity = std::min(capacity, available_capacity(sub_id));
+  }
+
+  for (const auto sub_id : publisher_it->second.take_ownership_subscriptions) {
+    capacity = std::min(capacity, available_capacity(sub_id));
+  }
+
+  return capacity;
```

Comment:
there could be possibility that this returns `std::numeric_limits<size_t>::max()`. is that expected behavior? Or if `publisher_it` can be found, there must be corresponding subscription id in the list?

Reply:
It's expected behavior. If not subscriptions are present, the buffer is "free" -> return max. value. To check for the number of (intra-process) subscriptions there is another method.

Reply:
so that means that publisher actually can write the maximum number of data in the buffer? i believe that what it means to return the maximum number here as interface, and what user application expects.